core/src/tile/tileManager.cpp

```diff
@@ -3,237 +3,350 @@
+        {
+            auto setTilesIter = m_tileSet.begin();
+            auto visTilesIter = visibleTiles.begin();
+
+            while (visTilesIter != visibleTiles.end()) {
+
+                auto& visTileId = *visTilesIter;
+                auto& curTileId = setTilesIter == m_tileSet.end() ? NOT_A_TILE : setTilesIter->first;
+
+                if (visTileId == curTileId) {
+                    if (setTilesIter->second->state() == MapTile::None) {
+                        logMsg("[%d, %d, %d] - Enqueue\n", curTileId.z, curTileId.x, curTileId.y);
+                        m_loadQueue.push_back(curTileId);
+                    }
+
+                    // tiles in both sets match
+                    ++setTilesIter;
+                    ++visTilesIter;
+
+                } else if (curTileId == NOT_A_TILE || visTileId < curTileId) {
+                    // tileSet is missing an element present in visibleTiles
+                    addTile(visTileId);
+                    ++visTilesIter;
+
                 } else {
+                    // visibleTiles is missing an element present in tileSet (handled below)
                     ++setTilesIter;
                 }
-            } else if (*visTilesIter < setTilesIter->first) {
-                // tileSet is missing an element present in visibleTiles (shouldn't occur)
-                ++visTilesIter;
-            } else {
-                // tiles in both sets match, move on
-                ++setTilesIter;
-                ++visTilesIter;
+            }
+        }
+
+        // Loop over tileSet and remove any tiles that are neither visible nor proxies
+        {
+            auto setTilesIter = m_tileSet.begin();
+            auto visTilesIter = visibleTiles.begin();
+
+            while (setTilesIter != m_tileSet.end()) {
+
+                auto& visTileId = visTilesIter == visibleTiles.end() ? NOT_A_TILE : *visTilesIter;
+                auto& curTileId = setTilesIter->first;
+
+                if (visTileId == curTileId) {
+                    // tiles in both sets match
+                    ++setTilesIter;
+                    ++visTilesIter;
+
+                } else if (visTileId == NOT_A_TILE || curTileId < visTileId) {
+                    // remove element from tileSet not present in visibleTiles
+                    const auto& tile = setTilesIter->second;
+                    if (tile->getProxyCounter() <= 0) {
+                        removeTile(setTilesIter);
+                    } else {
+                        ++setTilesIter;
+                    }
+                } else {
+                    // tileSet is missing an element present in visibleTiles (shouldn't occur)
+                    ++visTilesIter;
+                }
+            }
+        }
+    }
+
+    if (!m_loadQueue.empty() && m_loadPending < MAX_PARALLEL_DOWNLOADS) {
+        glm::dvec2 center(m_view->getPosition().x, -m_view->getPosition().y);
+
+        m_loadQueue.sort([&](const TileID& a, const TileID& b) {
```

Comment:
No, this part is not optimized yet and wip. Instead of pushing MAX_PARALLEL_DOWNLOADS to the backend  I would like to change it so that the loadQueue is polled by the platform download workers. I'm not sure if this should be done at the moment because it will require changes in all platform implementations.

A quick check on my laptop gave a maximum of 0.000322 ms for sorting 80 tiles (usually there are less tiles when the view is not fully tilted or tiles are partially loaded ~ 0.000020 ms). It is likely faster to just pick the tiles with the lowest distance though.